### ISSUE TITLE
lumen on linux needs to explicitly link in math function

### ISSUE THREAD

**Issue by KronicDeth** [lumen/lumen#552]
From https://github.com/lumen/lumen/pull/550/checks?check_run_id=1025707410

```
d: error: undefined symbol: trunc
>>> referenced by f64.rs:106 (/rustc/769d12eec1d0bc3708841dfc149d3ec98b04bec6/src/libstd/f64.rs:106)
>>>               liblumen_alloc-72f7f4b9cbcb0164.mu4e0zlce1yehj6.rcgu.o:(std::f64::_$LT$impl$u20$f64$GT$::trunc::h6853f0b8fb06f478) in archive /__w/lumen/lumen/bin/x86_64-unknown-linux-gnu/lib/lumenlib/x86_64-unknown-linux-gnu/lib/liblumen_rt_minimal.a

ld: error: undefined symbol: ceil
>>> referenced by f64.rs:67 (/rustc/769d12eec1d0bc3708841dfc149d3ec98b04bec6/src/libstd/f64.rs:67)
>>>               num_bigint-8ab64b69588169ec.num_bigint.86qajup1-cgu.11.rcgu.o:(std::f64::_$LT$impl$u20$f64$GT$::ceil::hfcf12885f89d2137) in archive /__w/lumen/lumen/bin/x86_64-unknown-linux-gnu/lib/lumenlib/x86_64-unknown-linux-gnu/lib/liblumen_rt_minimal.a
>>> referenced by f64.rs:67 (/rustc/769d12eec1d0bc3708841dfc149d3ec98b04bec6/src/libstd/f64.rs:67)
>>>               num_cpus-e786053c7f54cbbf.num_cpus.9ac0rxrr-cgu.1.rcgu.o:(std::f64::_$LT$impl$u20$f64$GT$::ceil::hf99d432f269a9eb6) in archive /__w/lumen/lumen/bin/x86_64-unknown-linux-gnu/lib/lumenlib/x86_64-unknown-linux-gnu/lib/liblumen_rt_minimal.a

ld: error: undefined symbol: log2
>>> referenced by f64.rs:445 (/rustc/769d12eec1d0bc3708841dfc149d3ec98b04bec6/src/libstd/f64.rs:445)
>>>               num_bigint-8ab64b69588169ec.num_bigint.86qajup1-cgu.11.rcgu.o:(std::f64::_$LT$impl$u20$f64$GT$::log2::_$u7b$$u7b$closure$u7d$$u7d$::he5f1438ede037144) in archive /__w/lumen/lumen/bin/x86_64-unknown-linux-gnu/lib/lumenlib/x86_64-unknown-linux-gnu/lib/liblumen_rt_minimal.a
clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by KronicDeth** [comment]
@bitwalker I can't remember, where do I change the libraries that are automatically linked (like the default `-l`'s) on a per target-basis?  I know we did this before, but Linux needs `-lm` too.

**Comment by KronicDeth** [comment]
Fixed by #549.
